lib/ansible/modules/remote_management/manageiq/manageiq_alert_profiles.py

```diff
@@ -0,0 +1,315 @@
+
+short_description: Configuration of alert profiles for ManageIQ
+extends_documentation_fragment: manageiq
+version_added: '2.5'
+author: Elad Alfassa (ealfassa@redhat.com)
+description:
+  - The manageiq_alert_profiles module supports listing, adding, updating and deleting alert profiles in ManageIQ.
+
+options:
+  state:
+    description:
+      - absent - alert profile should not exist,
+      - present - alert profile should exist,
+      - list - return a list of alert profiles.
+    choices: ['absent', 'present', 'list']
+    default: 'present'
+  name:
+    description:
+      - The unique alert profile name in ManageIQ.
+      - Required when state is "absent" or "present".
+  resource_type:
+    description:
+      - The resource type for the alert profile in ManageIQ. Required when state is "present".
+    choices: ['Vm', 'ContainerNode', 'MiqServer', 'Host', 'Storage', 'EmsCluster',
+              'ExtManagementSystem', 'MiddlewareServer']
+  alerts:
+    description:
+      - List of alert descriptions to assign to this profile.
+      - Required if state is "present"
+  notes:
+    description:
+      - Optional notes for this profile
+
+'''
+
+EXAMPLES = '''
+- name: List alert profiles in ManageIQ
+  manageiq_alert_profiles:
+    state: list
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Add an alert profile to ManageIQ
+  manageiq_alert_profiles:
+    state: present
+    name: Test profile
+    resource_type: ContainerNode
+    alerts:
+      - Test Alert 01
+      - Test Alert 02
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete an alert profile from ManageIQ
+  manageiq_alert_profiles:
+    state: absent
+    name: Test profile
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+class ManageIQAlertProfiles(object):
+    """ Object to execute alert profile management operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+        self.url = '{api_url}/alert_definition_profiles'.format(api_url=self.api_url)
+
+    def get_profiles(self):
+        """ Get all alert profiles from ManageIQ
+        """
+        try:
+            response = self.client.get(self.url + '?expand=alert_definitions,resources')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query alert profiles: {error}".format(error=e))
+        return response.get('resources') or []
```

Comment:
python nitpick: if you just want to return [] for missing 'resource' key, write `.get('resources', [])`.
As currently written is effectively `.get('resource', None) or []`, which will also accept `"resources": null` and other any "falsey" values like `"resources": ""`.  Harmless but one day someone will wonder "is this intentional, can server return non-arrays here?"...

Reply:
oops this is exactly opposite to what @resmo recommended previously https://github.com/ansible/ansible/pull/32354#discussion_r148456895
I don't really care, LGTM from me on either form.